File: Backend/ServerBackend/src/main/java/com/fran/dev/potjera/serverbackend/models/gamesession/playervhunter/PlayerVHunterBoardState.java
package com.fran.dev.potjera.serverbackend.models.gamesession.playervhunter;

import lombok.Builder;
import lombok.Data;
import org.jspecify.annotations.Nullable;

@Data
@Builder
public class PlayerVHunterBoardState {
    Boolean questionsStarted;
    Integer currentQuestionIndex;           // ← tracks position in question list
    @Nullable BoardQuestion boardQuestion;  // ← only current question (sent over network)
    @Nullable String hunterAnswer;
    @Nullable String playerAnswer;
    Integer hunterCorrectAnswers;
    Integer playerCorrectAnswers;
    Integer playerStartingIndex;
    Float moneyInGame;
    BoardPhase boardPhase;
}

File: AndroidApp/app/src/main/java/com/fran/dev/potjera/android/app/game/models/state/PlayerVHunterBoardState.kt
package com.fran.dev.potjera.android.app.game.models.state

import com.fran.dev.potjera.android.app.game.models.BoardQuestion
import com.fran.dev.potjera.android.app.game.models.enums.BoardPhase

data class PlayerVHunterBoardState(
    var questionsStarted: Boolean = false,
    var boardQuestion: BoardQuestion? = null,
    var hunterAnswer: String? = null,
    var playerAnswer: String? = null,
    var hunterCorrectAnswers: Int = 0,
    var playerCorrectAnswers: Int = 0,
    var playerStartingIndex: Int = 2,
    var moneyInGame: Float = 0f,
    var boardPhase: BoardPhase = BoardPhase.HUNTER_MAKING_OFFER
)
